test.js

```diff
@@ -440,6 +462,66 @@ describe('components', function () {
     })
   })
 
+  describe('es6 components', function () {
+    it('should work without args', function () {
+      out = mq(ES6Component)
+      out.should.have('aside')
+      out.should.contain('hello')
+    })
+
+    it('should work with directly injected components', function () {
+      out = mq(ES6Component, {data: 'my super data'})
+      out.should.have('aside')
+      out.should.contain('my super data')
+    })
+
+    it('should work without oninit', function () {
+      class SimpleES6Component {
+        view (vnode) {
+          return m('div', 'Hello from ' + vnode.attrs.name)
+        }
+      }
+      out = mq(SimpleES6Component, { name: 'Homer' })
+      out.should.have('div:contains(Hello from Homer)')
+    })
+
+    it('should call onremove on globalonremove', function (done) {
+      ES6Component.prototype.onremove = function () { done() }
+      var out = mq(ES6Component)
+      out.onremove()
+    })
+  })
+
+  describe('es6 instantiated component', function () {
+    it('should work without args', function () {
+      out = mq(new ES6Component())
+      out.should.have('aside')
+      out.should.contain('hello')
+    })
+
+    it('should work with directly injected components', function () {
+      out = mq(new ES6Component(), {data: 'my super data'})
+      out.should.have('aside')
+      out.should.contain('my super data')
+    })
+
+    it('should work without oninit', function () {
+      class SimpleES6Component {
+        view (vnode) {
+          return m('div', 'Hello from ' + vnode.attrs.name)
+        }
+      }
+      out = mq(new SimpleES6Component(), { name: 'Homer' })
+      out.should.have('div:contains(Hello from Homer)')
+    })
+
+    it('should call onremove on globalonremove', function (done) {
+      ES6Component.prototype.onremove = function () { done() }
+      var out = mq(new ES6Component())
+      out.onremove()
+    })
+  })
```

Comment:
Awesome work! Thanks a lot @jmooradi!

What I miss are some test cases where the same class component is rendered multiple times and the state of it persists. Like
```
var out = mq(new ES6Component())
// do something that changes the state of the component
// check if the component rendered the updated state
out.redraw()
// check if the component still rendered the updated state
```

Or even create a component with a click-handler that mutates it's own state, like a switch or something.

Also what I miss are some tests where a class component is embedded somewhere in the tree, at best als a class component that is embedded in another class component.

But nevertheless, great work, highly appreciated!

Reply:
@jmooradi any progress here?

Reply:
@jmooradi ping